content/guides/data-fetching.mdx

```diff
@@ -0,0 +1,158 @@
+# Data Fetching
+
+For most modern web applications, data fetching is a common task. 
+Solid has a built-in utility, `createResource` , that was created to simplify data fetching.
+
+## What is `createResource` ?
+
+`createResource` is a specialized [signal](/essentials/signal) designed specifically for managing asynchronous data fetching. 
+It wraps around the async operations, providing a way to handle various states: loading, success, and error.
+
+This function is non-blocking, meaning that `createResource` guarantees that the application remains responsive, even during the retrieval of information. 
+Because of this, common pitfalls of traditional async handling, such as unresponsive UIs during data fetching can be avoided.
+
+## Using `createResource`
+
+`createResource` requires a function that returns a promise as its argument. 
+Upon the call, `createResource` returns a signal which has reactive properties like loading, error, latest, etc. 
+These properties can be used to conditionally render JSX based on the current reactive state:
+
+```jsx
+import { createResource, Switch, Match, Show } from "solid-js";
+
+const fetchUser = async () =>
+ {
+  const response = await fetch(`https://swapi.dev/api/people/`);
+  return response.json();
+  }
+
+function App() {
+
+  const [user, { loading, error}] = createResource(fetchUser);
```

Comment:
The loading and error properties are properties of the user signal, not the second array object. Here's what this should look like:
```ts
const [user, { refetch }] = createResource(fetchUser);
```
refetch and mutate are the only properties returned in the second array object